CMakeLists.txt

```diff
@@ -275,6 +275,8 @@ OCV_OPTION(WITH_EIGEN "Include Eigen2/Eigen3 support" (NOT CV_DISABLE_OPTIMIZATI
 OCV_OPTION(WITH_FFMPEG "Include FFMPEG support" (NOT ANDROID)
   VISIBLE_IF NOT IOS AND NOT WINRT
   VERIFY HAVE_FFMPEG)
+OCV_OPTION(OPENCV_FFMPEG_ENABLE_LIBAVDEVICE "Include FFMPEG/libavdevice library support." OFF
+  VISIBLE_IF WITH_FFMPEG)
```

Comment:
Lets move it to the top of `modules/videoio/cmake/detect_ffmpeg.cmake`

This list is for high level configuration, "fine" tuning should go separately.

Reply:
Done :)